src/main/scala/mesosphere/marathon/NonEmptyIterable.scala

```diff
@@ -0,0 +1,17 @@
+package mesosphere.marathon
+
+object NonEmptyIterable {
+
+  def apply[A](head: A, tail: Iterable[A] = Iterable.empty[A]): NonEmptyIterable[A] = new NonEmptyIterable(head, tail)
+
+  def unapply[A, I <: Iterable[A]](iter: I with Iterable[A]): Option[NonEmptyIterable[A]] = {
+    if (iter.nonEmpty) Some(NonEmptyIterable(iter.head, iter.tail))
+    else None
+  }
+}
+
+case class NonEmptyIterable[A](override val head: A, override val tail: Iterable[A]) extends Iterable[A] {
+  override def iterator: Iterator[A] = Iterator.apply(head) ++ tail.iterator
+
+  override def headOption: Option[A] = Some(head)
```

Comment:
On the other site, `head` should be used - otherwise users will either endup treating a non-existing match case, or `get`-ing an `Option` and both doesn't make much sense for a `NonEmptyIterable`. On the other side it's simple and you don't have to throw an exception...